test/e2e/specs/interactivity/instant-search.spec.ts

```diff
@@ -0,0 +1,715 @@
+		await requestUtils.createPost( {
+			title: 'Fourth Test Post',
+			content: 'This is the fourth test post content.',
+			status: 'publish',
+			date_gmt: new Date(
+				new Date().getTime() - 1000 * 60 * 60 * 24 * 4
+			).toISOString(),
+		} );
+		await requestUtils.createPost( {
+			title: 'Third Test Post',
+			content: 'This is the third test post content.',
+			status: 'publish',
+			date_gmt: new Date(
+				new Date().getTime() - 1000 * 60 * 60 * 24 * 3
+			).toISOString(),
+		} );
+		await requestUtils.createPost( {
+			title: 'Second Test Post',
+			content: 'This is the second test post content.',
+			status: 'publish',
+			date_gmt: new Date(
+				new Date().getTime() - 1000 * 60 * 60 * 24 * 2
+			).toISOString(),
+		} );
+		await requestUtils.createPost( {
+			title: 'First Test Post',
+			content: 'This is the first test post content.',
+			status: 'publish',
+			date_gmt: new Date(
+				new Date().getTime() - 1000 * 60 * 60 * 24 * 1
+			).toISOString(),
+		} );
+
+		// Set the Blog pages show at most 2 posts
+		await requestUtils.updateSiteSettings( {
+			posts_per_page: 2,
+		} );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+		await requestUtils.activateTheme( 'twentytwentyone' );
+
+		// disable the gutenberg-search-query-block experiment
+		await requestUtils.setGutenbergExperiments( [] );
+	} );
+
+	test.describe( 'Custom Query', () => {
+		let pageId: number;
+
+		const queryId = 123;
+
+		test.beforeAll( async ( { requestUtils } ) => {
+			// Create page with custom query
+			const { id } = await requestUtils.createPage( {
+				status: 'publish',
+				date_gmt: new Date().toISOString(),
+				title: 'Custom Query',
+				content: `
+<!-- wp:query {"enhancedPagination":true,"queryId":${ queryId },"query":{"inherit":false,"perPage":2,"order":"desc","orderBy":"date","offset":0}} -->
+  <div class="wp-block-query" data-testid="custom-query">
+		<!-- wp:search {"label":"","buttonText":"Search"} /-->
+		<!-- wp:post-template -->
+			<!-- wp:post-title {"level":3} /-->
+			<!-- wp:post-excerpt /-->
+		<!-- /wp:post-template -->
+		<!-- wp:query-pagination -->
+			<!-- wp:query-pagination-previous /-->
+			<!-- wp:query-pagination-numbers /-->
+			<!-- wp:query-pagination-next /-->
+		<!-- /wp:query-pagination -->
+		<!-- wp:query-no-results -->
+			<!-- wp:paragraph -->
+			<p>No results found.</p>
+			<!-- /wp:paragraph -->
+		<!-- /wp:query-no-results -->
+	</div>
+<!-- /wp:query -->`,
+			} );
+
+			pageId = id;
+		} );
+
+		test.beforeEach( async ( { page } ) => {
+			await page.goto( `/?p=${ pageId }` );
+		} );
+
+		test( 'should update search results without page reload', async ( {
+			page,
+		} ) => {
+			// Check that the first post is shown initially
+			await expect(
+				page.getByText( 'First Test Post', { exact: true } )
+			).toBeVisible();
+
+			// Type in search input and verify results update
+			await page.locator( 'input[type="search"]' ).fill( 'Unique' );
```

Comment:
```suggestion
			await page.getByRole( 'searchbox' ).fill( 'Unique' );
```

Use accessible selectors whenever possible instead of `page.locator`.

Reply:
Fixed in 634e852779ab9a11396b34811de3ea5b8f174a9c